pkg/provisioners/channel_util.go

```diff
@@ -107,12 +123,16 @@ func createK8sService(ctx context.Context, client runtimeClient.Client, getSvc g
 	} else if err != nil {
 		return nil, err
 	}
-
 	// spec.clusterIP is immutable and is set on existing services. If we don't set this
 	// to the same value, we will encounter an error while updating.
-	svc.Spec.ClusterIP = current.Spec.ClusterIP
+	if svc.Spec.Type != corev1.ServiceTypeExternalName {
+		svc.Spec.ClusterIP = current.Spec.ClusterIP
+	}
 	if !equality.Semantic.DeepDerivative(svc.Spec, current.Spec) ||
-		!expectedLabelsPresent(current.ObjectMeta.Labels, svc.ObjectMeta.Labels) {
+		!expectedLabelsPresent(current.ObjectMeta.Labels, svc.ObjectMeta.Labels) ||
+		// This DeepEqual is necessary to force update dispatcher services when upgrading from 0.5 to 0.6.
```

Comment:
Will this be necessary long term? Or just for one release? If only for one release, add to the comment about when it can be removed.

Reply:
this is just for one release. I'll add it to the comment.